File: ios/LedgerlyRN/CategoryBadgeViewManager.m
#import "CategoryBadgeViewManager.h"
#import "CategoryBadgeView.h"

@implementation CategoryBadgeViewManager

RCT_EXPORT_MODULE(CategoryBadgeView)

- (UIView *)view {
  return [[CategoryBadgeView alloc] init];
}

RCT_EXPORT_VIEW_PROPERTY(category, NSString)
RCT_EXPORT_VIEW_PROPERTY(badgeColor, NSString)

@end

File: android/app/src/main/java/com/ledgerlyrn/CategoryBadgeViewManager.kt
package com.ledgerlyrn

import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class CategoryBadgeViewManager : SimpleViewManager<CategoryBadgeView>() {

  override fun getName() = "CategoryBadgeView"

  override fun createViewInstance(context: ThemedReactContext): CategoryBadgeView {
    return CategoryBadgeView(context)
  }

  @ReactProp(name = "category")
  fun setCategory(view: CategoryBadgeView, category: String?) {
    category?.let { view.setCategory(it) }
  }

  @ReactProp(name = "badgeColor")
  fun setBadgeColor(view: CategoryBadgeView, color: String?) {
    color?.let { view.setBadgeColor(it) }
  }
}